src/Nethermind/Nethermind.Consensus/ExecutionRequests/ExecutionRequestProcessor.cs

```diff
@@ -125,31 +130,35 @@ private void DecodeDepositRequest(LogEntry log, Span<byte> buffer)
         }
     }
 
-    private void ReadRequests(Block block, IWorldState state, IReleaseSpec spec, Address contractAddress, ArrayPoolList<byte[]> requests)
+    private void ReadRequests(Block block, IWorldState state, IReleaseSpec spec, Address contractAddress, ArrayPoolList<byte[]> requests,
+        Transaction systemTx, ExecutionRequestType type, string contractEmptyError, string contractFailedError)
     {
-        bool isWithdrawalRequests = contractAddress == spec.Eip7002ContractAddress;
-
-        int requestsByteSize = isWithdrawalRequests ? ExecutionRequestExtensions.WithdrawalRequestsBytesSize : ExecutionRequestExtensions.ConsolidationRequestsBytesSize;
-
-        if (!(isWithdrawalRequests ? spec.WithdrawalRequestsEnabled : spec.ConsolidationRequestsEnabled) || !state.AccountExists(contractAddress))
-            return;
+        if (!state.HasCode(contractAddress))
+        {
+            throw new InvalidBlockException(block, contractEmptyError);
+        }
 
         CallOutputTracer tracer = new();
 
-        _transactionProcessor.Execute(isWithdrawalRequests ? _withdrawalTransaction : _consolidationTransaction, new BlockExecutionContext(block.Header, spec), tracer);
+        _transactionProcessor.Execute(systemTx, new BlockExecutionContext(block.Header, spec), tracer);
```

Comment:
btw we don't need tracer, we could just use TransactionResult that is being returned

Reply:
We need ReturnValue

Reply:
still you can use error code